qubesappmenusext/__init__.py

```diff
@@ -88,6 +88,9 @@ def on_feature_del_appmenus_dispvm(self, vm, event, feature):
     @qubes.ext.handler('domain-feature-set:appmenus-dispvm')
     def on_feature_set_appmenus_dispvm(self, vm, event, feature,
             value, oldvalue=None):
+        if not vm.template_for_dispvms:
```

Comment:
1. Blocking a change should be done in `domain-feature-pre-set:...` event (here it's too late, you'll just interrupt reacting to the change, but it's already made).
2. This should check what value the feature is going to be set. Specifically, it should not reject setting to false value.